src/mumble/AudioOutput.cpp

```diff
@@ -585,38 +600,51 @@ bool AudioOutput::mix(void *outbuff, unsigned int nsamp) {
 					const float str = svol[s] * calcGain(dot, len) * volumeAdjustment;
 					float * RESTRICT o = output + s;
 					const float old = (aop->pfVolume[s] >= 0.0f) ? aop->pfVolume[s] : str;
-					const float inc = (str - old) / static_cast<float>(nsamp);
+					const float inc = (str - old) / static_cast<float>(frameNumber);
 					aop->pfVolume[s] = str;
 					/*
 										qWarning("%d: Pos %f %f %f : Dot %f Len %f Str %f", s, speaker[s*3+0], speaker[s*3+1], speaker[s*3+2], dot, len, str);
 					*/
 					if ((old >= 0.00000001f) || (str >= 0.00000001f))
-						for (unsigned int i=0;i<nsamp;++i)
-							o[i*nchan] += pfBuffer[i] * (old + inc*static_cast<float>(i));
+                        for (unsigned int i=0;i<frameNumber;++i) {
+                            if (speech && speech->bStereo) {
+                                // Mix stereo user's stream into mono
+                                o[i * nchan] += (pfBuffer[2*i] / 2.0 + pfBuffer[2*i+1] / 2.0) * (old + inc * static_cast<float>(i));
+                            } else {
+                                o[i * nchan] += pfBuffer[i] * (old + inc * static_cast<float>(i));
+                            }
+                        }
 				}
```

Comment:
The indentation here is off (either the `if` above is indented too far or this block is not indented enough :thinking:

Reply:
Sorry I messed up the indentation with space and tab again T_T